pkg/kubectl/cmd/delete.go

```diff
@@ -150,7 +150,7 @@ func NewCmdDelete(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
 	usage := "containing the resource to delete."
 	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
 	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on.")
-	cmd.Flags().BoolVar(&options.DeleteAll, "all", false, "Select all resources in the namespace of the specified resource types.")
+	cmd.Flags().BoolVar(&options.DeleteAll, "all", false, "Select all resources (include the uninitialized objects by default, unless user explicitly set --include-uninitialized=false) in the namespace of the specified resource types.")
```

Comment:
Delete all resources in the namespace of the specified resource types. It will delete the uninitialized objects as well, unless --include-uninitialized=false is set explicitly.